lib/transfer_util.cu

```diff
@@ -704,23 +702,24 @@ namespace quda {
 
   template<typename Float>
   void BlockOrthogonalize(ColorSpinorField &V, int Nvec, 
```

Comment:
Maybe I'm misunderstanding your question, but there isn't a perfect one to one mapping between nSpin and spinBlockSize. 

Maybe this helps: the ultimate culprit line that I thought required this extra template parameter is:

(644:) constexpr int coarseSpin = (nSpin == 4 || nSpin == 2 || spinBlockSize == 0) ? 2 : 1;

Which used to just check (nSpin == 4 || nSpin == 2) for Wilson-like -> coarse and coarse -> coarse, respectively. nSpin == 1 wouldn't be unique to staggered (lets say if you wanted to coarsen the Laplace operator). If we want to assume nSpin == 1 is uniquely staggered, then I can throw all of this extra stuff out the window.

Odds are I'm just misunderstanding what a trait could be used for, though!